[BUG] [RUST Server] Invalid validation generated for large integers after updating to 7.15

#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

After updating the generator to `7.15`, building our Rust Server fails to compile due to invalid validation rules being generated for large integers.

##### openapi-generator version

The generation works fine on both version however, after updating to 7.15, the generated code does not compile and fails with the following error:

```
error[E0600]: cannot apply unary operator `-` to type `u64`
   --> src/models.rs:887:34
    |
887 |             range(min = 0, max = -8446744073709551616),
    |                                  ^ cannot apply unary operator `-`
    |
    = note: unsigned values cannot be negated
```

##### OpenAPI declaration file content or url

```
openapi: 3.0.0

info:
  title: 'Example'
  description: 'test'
  version: 0.0.1
  contact:
    email: 'info@pietrobongiovanni.com'

servers:
  - url: '{protocol}://{hostname}:{port}/api/v1'
    variables:
      protocol:
        enum:
          - http
          - https
        default: http
      hostname:
        default: 'localhost'
      port:
        default: '8080'

components:

  schemas:
    paginationResponse:
      type: object
      properties:
        page:
          type: integer
          minimum: 0
          maximum: 10000000000000000000
          example: 0
        pageSize:
          type: integer
          minimum: 1
          maximum: 10000000000000000000
          example: 100
      required:
        - page
        - pageSize

    deviceId:
      type: string
      description: 'Device unique identifier'
      example: 'device_01jrd1sbzpfg89kqzcfazy6d0q'

    device:
      allOf:
        - type: object
          required:
            - id
          properties:
            id:
              $ref: '#/components/schemas/deviceId'

    paginatedDevices:
      type: object
      required:
        - items
        - pagination
      properties:
        items:
          type: array
          items:
            $ref: '#/components/schemas/device'
        pagination:
          $ref: '#/components/schemas/paginationResponse'

paths:
  /devices:
    summary: 'Devices'
    get:
      operationId: getDevices
      responses:
        '200':
          description: '(200) OK'
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/paginatedDevices'
```

##### Generation Details

We generate the code in a podman container. Generation works fine but the subsequent cargo build of the server fails. Please see this `generate.sh`:

```
#!/bin/bash

OPENAPI_GENERATOR_VERSION=v7.14.0

set -x

rm -rf generated/

podman run --rm -v "$(pwd)":/repo:Z openapitools/openapi-generator-cli:$OPENAPI_GENERATOR_VERSION \
    generate \
    -i /repo/openapi.yaml \
    -g rust-server \
    -o /repo/generated/rust-server \
    --model-name-prefix api \
    --additional-properties hideGenerationTimestamp=true

pushd "$(pwd)/generated/rust-server" > /dev/null
echo
echo "Building rust-server library..."
echo
cargo build
echo
echo "Building rust-server example client..."
echo
cargo build --example client
echo
echo "Building rust-server example server..."
echo
cargo build --example server
popd > /dev/null
```

##### Steps to reproduce

Copy the above openapi.yaml and generate.sh in a folder and run `generate.sh`  to generate and build the rust server.
With `OPENAPI_GENERATOR_VERSION=v7.14.0` the server will compile successfully.
If you set `OPENAPI_GENERATOR_VERSION` to `v7.15.0` the compilation of the server will fail:

```
error[E0600]: cannot apply unary operator `-` to type `u64`
   --> src/models.rs:461:34
    |
461 |             range(min = 1, max = -8446744073709551616),
    |                                  ^ cannot apply unary operator `-`
    |
    = note: unsigned values cannot be negated

For more information about this error, try `rustc --explain E0600`.
```

--- a/modules/openapi-generator/src/main/java/org/openapitools/codegen/utils/ModelUtils.java
+++ b/modules/openapi-generator/src/main/java/org/openapitools/codegen/utils/ModelUtils.java
@@ -1941,7 +1941,7 @@ public class ModelUtils {
         if (multipleOf != null) target.setMultipleOf(multipleOf);
         if (minimum != null) {
             if (isIntegerSchema(schema)) {
-                target.setMinimum(String.valueOf(minimum.longValue()));
+                target.setMinimum(String.valueOf(minimum.toBigInteger()));
             } else {
                 target.setMinimum(String.valueOf(minimum));
             }
@@ -1949,7 +1949,7 @@ public class ModelUtils {
         }
         if (maximum != null) {
             if (isIntegerSchema(schema)) {
-                target.setMaximum(String.valueOf(maximum.longValue()));
+                target.setMaximum(String.valueOf(maximum.toBigInteger()));
             } else {
                 target.setMaximum(String.valueOf(maximum));
             }

--- a/modules/openapi-generator/src/main/resources/rust-server/models.mustache
+++ b/modules/openapi-generator/src/main/resources/rust-server/models.mustache
@@ -312,15 +312,15 @@ pub struct {{{classname}}} {
         {{/pattern}}
         {{#maximum}}
             {{#minimum}}
-            range(min = {{minimum}}, max = {{maximum}}),
+            range(min = {{minimum}}{{dataType}}, max = {{maximum}}{{dataType}}),
             {{/minimum}}
             {{^minimum}}
-            range(max = {{maximum}}),
+            range(max = {{maximum}}{{dataType}}),
             {{/minimum}}
         {{/maximum}}
         {{#minimum}}
             {{^maximum}}
-            range(min = {{minimum}}),
+            range(min = {{minimum}}{{dataType}}),
             {{/maximum}}
         {{/minimum}}
         {{#maxItems}}

--- a/modules/openapi-generator/src/test/resources/3_0/rust-server/openapi-v3.yaml
+++ b/modules/openapi-generator/src/test/resources/3_0/rust-server/openapi-v3.yaml
@@ -693,6 +693,9 @@ components:
           type: boolean
         optionalParam:
           type: integer
+          minimum: 1
+          maximum: 10000000000000000000
+          example: 100
     ObjectHeader:
       type: object
       required:

--- a/samples/server/petstore/rust-server-deprecated/output/openapi-v3/api/openapi.yaml
+++ b/samples/server/petstore/rust-server-deprecated/output/openapi-v3/api/openapi.yaml
@@ -735,11 +735,14 @@ components:
     ObjectParam:
       example:
         requiredParam: true
-        optionalParam: 0
+        optionalParam: 100
       properties:
         requiredParam:
           type: boolean
         optionalParam:
+          example: 100
+          maximum: 10000000000000000000
+          minimum: 1
           type: integer
       required:
       - requiredParam

--- a/samples/server/petstore/rust-server-deprecated/output/openapi-v3/docs/ObjectParam.md
+++ b/samples/server/petstore/rust-server-deprecated/output/openapi-v3/docs/ObjectParam.md
@@ -4,7 +4,7 @@
 Name | Type | Description | Notes
 ------------ | ------------- | ------------- | -------------
 **required_param** | **bool** |  | 
-**optional_param** | **i32** |  | [optional] [default to None]
+**optional_param** | **u64** |  | [optional] [default to None]
 
 [[Back to Model list]](../README.md#documentation-for-models) [[Back to API list]](../README.md#documentation-for-api-endpoints) [[Back to README]](../README.md)
 

--- a/samples/server/petstore/rust-server-deprecated/output/openapi-v3/src/models.rs
+++ b/samples/server/petstore/rust-server-deprecated/output/openapi-v3/src/models.rs
@@ -3597,8 +3597,11 @@ pub struct ObjectParam {
     pub required_param: bool,
 
     #[serde(rename = "optionalParam")]
+    #[validate(
+            range(min = 1, max = 10000000000000000000),
+        )]
     #[serde(skip_serializing_if="Option::is_none")]
-    pub optional_param: Option<i32>,
+    pub optional_param: Option<u64>,
 
 }
 
@@ -3645,7 +3648,7 @@ impl std::str::FromStr for ObjectParam {
         #[allow(dead_code)]
         struct IntermediateRep {
             pub required_param: Vec<bool>,
-            pub optional_param: Vec<i32>,
+            pub optional_param: Vec<u64>,
         }
 
         let mut intermediate_rep = IntermediateRep::default();
@@ -3666,7 +3669,7 @@ impl std::str::FromStr for ObjectParam {
                     #[allow(clippy::redundant_clone)]
                     "requiredParam" => intermediate_rep.required_param.push(<bool as std::str::FromStr>::from_str(val).map_err(|x| x.to_string())?),
                     #[allow(clippy::redundant_clone)]
-                    "optionalParam" => intermediate_rep.optional_param.push(<i32 as std::str::FromStr>::from_str(val).map_err(|x| x.to_string())?),
+                    "optionalParam" => intermediate_rep.optional_param.push(<u64 as std::str::FromStr>::from_str(val).map_err(|x| x.to_string())?),
                     _ => return std::result::Result::Err("Unexpected key while parsing ObjectParam".to_string())
                 }
             }

--- a/samples/server/petstore/rust-server/output/openapi-v3/api/openapi.yaml
+++ b/samples/server/petstore/rust-server/output/openapi-v3/api/openapi.yaml
@@ -735,11 +735,14 @@ components:
     ObjectParam:
       example:
         requiredParam: true
-        optionalParam: 0
+        optionalParam: 100
       properties:
         requiredParam:
           type: boolean
         optionalParam:
+          example: 100
+          maximum: 10000000000000000000
+          minimum: 1
           type: integer
       required:
       - requiredParam

--- a/samples/server/petstore/rust-server/output/openapi-v3/docs/ObjectParam.md
+++ b/samples/server/petstore/rust-server/output/openapi-v3/docs/ObjectParam.md
@@ -4,7 +4,7 @@
 Name | Type | Description | Notes
 ------------ | ------------- | ------------- | -------------
 **required_param** | **bool** |  | 
-**optional_param** | **i32** |  | [optional] [default to None]
+**optional_param** | **u64** |  | [optional] [default to None]
 
 [[Back to Model list]](../README.md#documentation-for-models) [[Back to API list]](../README.md#documentation-for-api-endpoints) [[Back to README]](../README.md)
 

--- a/samples/server/petstore/rust-server/output/openapi-v3/src/models.rs
+++ b/samples/server/petstore/rust-server/output/openapi-v3/src/models.rs
@@ -3597,8 +3597,11 @@ pub struct ObjectParam {
     pub required_param: bool,
 
     #[serde(rename = "optionalParam")]
+    #[validate(
+            range(min = 1u64, max = 10000000000000000000u64),
+        )]
     #[serde(skip_serializing_if="Option::is_none")]
-    pub optional_param: Option<i32>,
+    pub optional_param: Option<u64>,
 
 }
 
@@ -3645,7 +3648,7 @@ impl std::str::FromStr for ObjectParam {
         #[allow(dead_code)]
         struct IntermediateRep {
             pub required_param: Vec<bool>,
-            pub optional_param: Vec<i32>,
+            pub optional_param: Vec<u64>,
         }
 
         let mut intermediate_rep = IntermediateRep::default();
@@ -3666,7 +3669,7 @@ impl std::str::FromStr for ObjectParam {
                     #[allow(clippy::redundant_clone)]
                     "requiredParam" => intermediate_rep.required_param.push(<bool as std::str::FromStr>::from_str(val).map_err(|x| x.to_string())?),
                     #[allow(clippy::redundant_clone)]
-                    "optionalParam" => intermediate_rep.optional_param.push(<i32 as std::str::FromStr>::from_str(val).map_err(|x| x.to_string())?),
+                    "optionalParam" => intermediate_rep.optional_param.push(<u64 as std::str::FromStr>::from_str(val).map_err(|x| x.to_string())?),
                     _ => return std::result::Result::Err("Unexpected key while parsing ObjectParam".to_string())
                 }
             }

--- a/samples/server/petstore/rust-server/output/petstore-with-fake-endpoints-models-for-testing/src/models.rs
+++ b/samples/server/petstore/rust-server/output/petstore-with-fake-endpoints-models-for-testing/src/models.rs
@@ -3857,14 +3857,14 @@ impl FindPetsByStatusStatusParameterInner {
 pub struct FormatTest {
     #[serde(rename = "integer")]
     #[validate(
-            range(min = 10, max = 100),
+            range(min = 10u8, max = 100u8),
         )]
     #[serde(skip_serializing_if="Option::is_none")]
     pub integer: Option<u8>,
 
     #[serde(rename = "int32")]
     #[validate(
-            range(min = 20, max = 200),
+            range(min = 20u32, max = 200u32),
         )]
     #[serde(skip_serializing_if="Option::is_none")]
     pub int32: Option<u32>,
@@ -3875,20 +3875,20 @@ pub struct FormatTest {
 
     #[serde(rename = "number")]
     #[validate(
-            range(min = 32.1, max = 543.2),
+            range(min = 32.1f64, max = 543.2f64),
         )]
     pub number: f64,
 
     #[serde(rename = "float")]
     #[validate(
-            range(min = 54.3, max = 987.6),
+            range(min = 54.3f32, max = 987.6f32),
         )]
     #[serde(skip_serializing_if="Option::is_none")]
     pub float: Option<f32>,
 
     #[serde(rename = "double")]
     #[validate(
-            range(min = 67.8, max = 123.4),
+            range(min = 67.8f64, max = 123.4f64),
         )]
     #[serde(skip_serializing_if="Option::is_none")]
     pub double: Option<f64>,

